addons/mail/static/src/core/web/messaging_menu.js

```diff
@@ -80,8 +81,12 @@ export class MessagingMenu extends Component {
     get hasPreviews() {
         return (
             this.threads.length > 0 ||
-            (this.store.failures.length > 0 && this.store.discuss.activeTab === "all") ||
-            (this.notification.permission === "prompt" && this.store.discuss.activeTab === "all")
+            (this.store.failures.length > 0 &&
+                this.store.discuss.activeTab === "main" &&
+                !this.env.inDiscussApp) ||
+            (this.notification.permission === "prompt" &&
+                this.store.discuss.activeTab === "main" &&
+                !this.env.inDiscussApp)
```

Comment:
the condition on `!this.env.inDiscussApp` could just be at the end I think, not repeated on each part of the `||`.

Reply:
Code sucks, but it can't be put at end because `this.threads.length > 0` should be `true` regardless of `env.inDiscussApp`.

This `activeTab === "main" && !env.inDiscussApp` matches the old "all" and not match old "mailbox".
This is just a way to not consider "previews" in mailbox, as this screen is quite special

Reply:
Ho I understand!
Well I think you still can avoid the duplication like this:
```js
    get hasPreviews() {
        return (
            this.threads.length > 0 || (
                (this.store.failures.length > 0 && this.store.discuss.activeTab === "all") ||
                (this.notification.permission === "prompt" && this.store.discuss.activeTab === "all")
                    (this.store.failures.length > 0 &&
                        this.store.discuss.activeTab === "main" &&
                        !this.env.inDiscussApp) ||
                (this.notification.permission === "prompt" &&
                    this.store.discuss.activeTab === "main"
                )
                && !this.env.inDiscussApp)
        );
    }

```